### PR TITLE
fix(list): associate to iteration from kebab menu

### PR DESCRIPTION
**Pull request by vikram-raj**
Fix https://github.com/openshiftio/openshift.io/issues/951

### CODE REVIEW

**Comment by fabric8cd** [comment]
@vikram-raj fabric8/fabric8-ui:SNAPSHOT-PR-2170-1 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2170-fabric8-planner.badger.fabric8.io

File: src/app/components/work-item-iteration-modal/work-item-iteration-modal.component.ts [hunk 1]
```diff
@@ -200,17 +207,28 @@ export class FabPlannerAssociateIterationModalComponent {
...
-        this.workItemService.emitEditWI(workItem);
+        this.workItemService.emitEditWI(this.workItem);
       });
   }
 
   open(event: any) {
     this.getIterations();
+    this.workItemPayload = {
+      id: this.workItem.id,
+      number: this.workItem.number,
```
Comment (sanbornsen) [inline src/app/components/work-item-iteration-modal/work-item-iteration-modal.component.ts]: How do you get number is **this.workItem** ?

File: src/app/components/work-item-iteration-modal/work-item-iteration-modal.component.ts [hunk 2]
```diff
@@ -200,17 +207,28 @@ export class FabPlannerAssociateIterationModalComponent {
...
+        self: this.workItem.links.self
+      },
+      type: this.workItem.type
+    };
     this.iterationAssociationModal.open();
   }
 
   actionOnOpen() {
-    if (this.workItem.relationships.iteration.data) {
+    if (this.workItem.relationships.iteration.data){
```
Comment (sanbornsen) [inline src/app/components/work-item-iteration-modal/work-item-iteration-modal.component.ts]: Why would you remove the space between ) and { ?

File: src/app/components/work-item-iteration-modal/work-item-iteration-modal.component.ts [hunk 3]
```diff
@@ -200,17 +207,28 @@ export class FabPlannerAssociateIterationModalComponent {
...
         this.workItem.relationships.iteration = workItem.relationships.iteration;
         this.workItem.attributes['version'] = workItem.attributes['version'];
-        this.workItemService.emitEditWI(workItem);
+        this.workItemService.emitEditWI(this.workItem);
       });
   }
 
   open(event: any) {
     this.getIterations();
+    this.workItemPayload = {
```
Comment (sanbornsen) [inline src/app/components/work-item-iteration-modal/work-item-iteration-modal.component.ts]: Why are you initiating payload in this function? workItem is an input to this component, should'd the payload be updated in **ngOnChanges** lifecycle hook? You'd want to update the payload as soon as the input changes with a work item.

File: src/app/components/work-item-iteration-modal/work-item-iteration-modal.component.ts (hunk 1 above)
Comment (vikram-raj) [reply src/app/components/work-item-iteration-modal/work-item-iteration-modal.component.ts]: Ah my bad will fix it.

**Comment by fabric8cd** [comment]
@vikram-raj fabric8/fabric8-ui:SNAPSHOT-PR-2170-2 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2170-fabric8-planner.badger.fabric8.io

File: src/app/components/work-item-iteration-modal/work-item-iteration-modal.component.ts (hunk 3 above)
Comment (vikram-raj) [reply src/app/components/work-item-iteration-modal/work-item-iteration-modal.component.ts]: It is modal so, it only required when user open the modal. So I don't think I have to update it in ngOnchange.

Comment (sanbornsen) [reply src/app/components/work-item-iteration-modal/work-item-iteration-modal.component.ts]: But in future, if you have to implement something case where while the modal is open, you need switch the work item and get the modal updated too, how would you do that? Life cycle hooks are there for a reason, and we should make use of it. Think about it, and even then you think it's okay to keep it here, I am good with that for now.

Comment (vikram-raj) [reply src/app/components/work-item-iteration-modal/work-item-iteration-modal.component.ts]: Changes in https://github.com/fabric8-ui/fabric8-planner/pull/2170/commits/28b5a9ec89bbd4f81ec7e93fcf13eb175359d18d

**Comment by fabric8cd** [comment]
@vikram-raj fabric8/fabric8-ui:SNAPSHOT-PR-2170-3 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2170-fabric8-planner.badger.fabric8.io

**Pull request merged**